src/xrc/xh_choicbk.cpp

```diff
@@ -123,9 +116,25 @@ wxObject *wxChoicebookXmlHandler::DoCreateResource()
         m_choicebook = nb;
         bool old_ins = m_isInside;
         m_isInside = true;
+        wxVector<wxBitmapBundle> old_images = m_bookImages;
+        m_bookImages.clear();
+        wxVector<size_t> old_imageIdx = m_bookImagesIdx;
+        m_bookImagesIdx.clear();
         CreateChildren(m_choicebook, true/*only this handler*/);
+
+        if ( !m_bookImages.empty() )
+        {
+            m_choicebook->SetImages(m_bookImages);
+            for ( size_t i = 0; i < m_bookImagesIdx.size(); ++i )
+            {
+                m_choicebook->SetPageImage( m_bookImagesIdx[i], i );
+            }
+        }
```

Comment:
This could be refactored into `wxBookCtrlXmlResourceHandlerBase::DoSetPageImages()` to avoid duplicating it in many places.